packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1641,6 +1682,7 @@ class BuildOwner {
       Timeline.finishSync();
       assert(_debugBuilding);
       assert(() {
+        _dirtyElementsNeedsResorting = null;
```

Comment:
Why in an assert?  It seems dangerous for this field to have different values in production and debug given that it is used in both configurations.  I'd just assign null between lines 1681 and 1682.